community/modules/scripts/htcondor-install/files/autoscaler.py

```diff
@@ -180,47 +181,30 @@ def scale(self):
         else:
             self.zoneargs = {"zone": self.zone}
 
-        # Get total number of jobs in the queue that includes number of jobs
-        # waiting as well as number of jobs already assigned to nodes
-        queue_length_req = (
-            'condor_q -allusers -totals -format "%d " Jobs -format "%d " Idle -format "%d " Held'
-        )
-        queue_length_resp = os.popen(queue_length_req).read().split()
+        # Count jobs that are running or idle ("potentially runnable"); do not
+        # count held jobs or jobs in transitioning states
+        # https://htcondor.readthedocs.io/en/latest/classad-attributes/job-classad-attributes.html#JobStatus
+        schedd = htcondor.Schedd()
+        job_attributes = [ "RequestCpus", "RequestMemory", "RequestGpus" ]
+        running_job_ads = schedd.query(constraint="JobStatus==2", projection=job_attributes)
+        idle_job_ads = schedd.query(constraint="JobStatus==1", projection=job_attributes)
 
-        if len(queue_length_resp) > 1:
-            queue = int(queue_length_resp[0])
-            idle_jobs = int(queue_length_resp[1])
-            on_hold_jobs = int(queue_length_resp[2])
-        else:
-            queue = 0
-            idle_jobs = 0
-            on_hold_jobs = 0
+        total_idle_request_cpus = sum(j["RequestCpus"] for j in idle_job_ads)
+        total_running_request_cpus = sum(j["RequestCpus"] for j in running_job_ads)
+        queue = total_idle_request_cpus + total_running_request_cpus
 
-        print("Total queue length: " + str(queue))
-        print("Idle jobs: " + str(idle_jobs))
-        print("Jobs on hold: " + str(on_hold_jobs))
+        print(f"Running job RequestCpus: {total_running_request_cpus}")
+        print(f"Idle job RequestCpus: {total_idle_request_cpus}")
```

Comment:
Will condor people understand the RequestCpus reference. For me this is less clear than the previous message (even though it is more specific) because it leaves me wondering what the reference to RequestCpus means.

Reply:
HTCondor admins will know `RequestCpus` (for reasons noted above) but it's easy for me to alter the text to something more completely correct: "Total CPUs requested by running/idle jobs".